packages/lodestar/src/sync/sync.ts

```diff
@@ -212,10 +214,13 @@ export class BeaconSync implements IBeaconSync {
     const blockRoot = this.config.types.BeaconBlock.hashTreeRoot(err.job.signedBlock.message);
     const unknownAncestorRoot = this.chain.pendingBlocks.getMissingAncestor(blockRoot);
     const missingRootHex = toHexString(unknownAncestorRoot);
-    if (this.processingRoots.has(missingRootHex)) {
+    const MAX_TIME_IN_CACHE = 60 * 1000;
+    const lastTimeInCache = this.processingRoots.get(missingRootHex);
+    // in case beaconBlocksByRoot timeout, we can retry after 1 minute
+    if (notNullish(lastTimeInCache) && Date.now() - lastTimeInCache < MAX_TIME_IN_CACHE) {
```

Comment:
maybe better to use some lru cache implementation or write our own generic one that can be used here

Reply:
gossipsub-js uses `time-cache`, we should use it here too